app/uk/gov/hmrc/agentassurance/models/AgencyDetails.scala

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.agentassurance.models
+
+import play.api.libs.json.{Json, OFormat}
+
+case class AgencyDetails(
+                          agencyName: Option[String],
+                          agencyEmail: Option[String],
+                          agencyTelephone: Option[String],
+                          agencyAddress: Option[BusinessAddress]
+                        )
```

Comment:
Looks bit suspicious.  Why all fields are optional ?

Reply:
Good question! This is the same model used in agent-client-authorisation https://github.com/hmrc/agent-client-authorisation/blob/main/app/uk/gov/hmrc/agentclientauthorisation/model/AgentDetailsDesResponse.scala

And we did find it annoying that all the fields are optional, probably needs a look into see if any can be made mandatory otherwise you can get the weird scenario where you get Some(AgencyDetails(None, None,None,None))!